File: solutions/javascript/RemoveLinkedListElements/RemoveLinkedListElements.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} val
 * @return {ListNode}
 */
var removeElements = function(head, val) {
    var l = new ListNode(0);
    l.next = head;
    head = l;
    while (l.next) {
        if (l.next.val == val) {
            l.next = l.next.next;
        } else {
            l = l.next;
        }
    }
    return head.next;
};

File: solutions/java/RemoveLinkedListElements/RemoveLinkedListElements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode l = new ListNode(0);
        l.next = head;
        head = l;
        while (null != l.next) {
            if (l.next.val == val) {
                l.next = l.next.next;
            } else {
                l = l.next;
            }
        }
        return head.next;
    }
}